cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -41,42 +68,97 @@
 @Singleton
 public class StreamSizeScheduler implements Scheduler {
   private static final Logger LOG = LoggerFactory.getLogger(StreamSizeScheduler.class);
+  private static final int STREAM_POLLING_THREAD_POOL_SIZE = 10;
 
+  private final long pollingDelay;
   private final NotificationService notificationService;
   private final StreamAdmin streamAdmin;
   private final StoreFactory storeFactory;
   private final ProgramRuntimeService programRuntimeService;
   private final PreferencesStore preferencesStore;
+  private final ConcurrentMap<Id.Stream, StreamSubscriber> streamSubscribers;
 
+  // Key is scheduleId
+  private final ConcurrentMap<String, StreamSubscriber> scheduleSubscribers;
+
+  private Store store;
+  private Executor notificationExecutor;
+  private ScheduledExecutorService streamPollingExecutor;
 
   @Inject
   public StreamSizeScheduler(CConfiguration cConf, NotificationService notificationService, StreamAdmin streamAdmin,
                              StoreFactory storeFactory, ProgramRuntimeService programRuntimeService,
                              PreferencesStore preferencesStore) {
+    this.pollingDelay = cConf.getLong(Constants.Notification.Stream.STREAM_SIZE_SCHEDULE_POLLING_DELAY);
```

Comment:
I will convert to millis only once, that's true.
We talked about that with @chtyim, and decided to only go with a global delay for now, and maybe later enable to configure a delay per schedule.